File: AdminPanel/AdminPanel.Web/Modules/Souqcom/Category/CategoryForm.cs
namespace AdminPanel.Souqcom.Forms;

[FormScript("Souqcom.Category")]
[BasedOnRow(typeof(CategoryRow), CheckNames = true)]
public class CategoryForm
{
    public string Name { get; set; }
    public string Photo { get; set; }
    public string Description { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Souqcom/CategoryForm.ts
import { initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface CategoryForm {
    Name: StringEditor;
    Photo: StringEditor;
    Description: StringEditor;
}

export class CategoryForm extends PrefixedContext {
    static readonly formKey = 'Souqcom.Category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoryForm.init) {
            CategoryForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoryForm, [
                'Name', w0,
                'Photo', w0,
                'Description', w0
            ]);
        }
    }
}